lib/Parse/Lexer.cpp

```diff
@@ -1887,11 +1887,17 @@ void Lexer::lexStringLiteral(unsigned CustomDelimiterLen) {
       } else if ((*CurPtr == '\r' || *CurPtr == '\n') && IsMultilineString) {
         // The only case we reach here is unterminated single line string in the
         // interpolation. For better recovery, go on after emitting an error.
-        diagnose(CurPtr, diag::lex_unterminated_string);
+        diagnose(CurPtr, diag::string_interpolation_unclosed);
         wasErroneous = true;
         continue;
       } else {
```

Comment:
This branch, as the comment mentions, is entered only when we find an unterminated string inside an interpolation, so it should also diagnose that.

Reply:
There, added that in the most recent commit